plugins/source/mysql/client/types.go

```diff
@@ -1,64 +1,146 @@
 package client
 
 import (
-	"github.com/cloudquery/plugin-sdk/v2/schema"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/cloudquery/plugin-sdk/v3/types"
 )
 
-func SQLType(t schema.ValueType) string {
-	switch t {
-	case schema.TypeBool:
+const defaultPrecision = 10
+const defaultScale = 0
+
+func getValue(str string, defaultValue int32) int32 {
+	val, err := strconv.ParseInt(str, 10, 32)
+	if err != nil {
+		return defaultValue
+	}
+	return int32(val)
+}
+
+func getPrecisionAndScale(dataType string) (precision, scale int32) {
+	str := strings.TrimPrefix(dataType, "decimal")
+	str = strings.TrimPrefix(str, "numeric")
+	if str == "" {
+		return defaultPrecision, defaultScale
+	}
+	str = strings.TrimPrefix(str, "(")
+	str = strings.TrimSuffix(str, ")")
+	parts := strings.Split(str, ",")
+
+	switch len(parts) {
+	case 1:
+		precision = getValue(parts[0], defaultPrecision)
+		scale = defaultScale
+	case 2:
+		precision = getValue(parts[0], defaultPrecision)
+		scale = getValue(parts[1], defaultScale)
+	default:
+		precision = defaultPrecision
+		scale = defaultScale
+	}
+	return precision, scale
+}
+
+func SQLType(t arrow.DataType) string {
+	switch v := t.(type) {
+	case *arrow.BooleanType:
 		return "bool"
-	case schema.TypeInt:
+	case *arrow.Int8Type:
+		return "tinyint"
+	case *arrow.Int16Type:
+		return "smallint"
+	case *arrow.Int32Type:
+		return "int"
+	case *arrow.Int64Type:
 		return "bigint"
-	case schema.TypeFloat:
+	case *arrow.Uint8Type:
+		return "tinyint unsigned"
+	case *arrow.Uint16Type:
+		return "smallint unsigned"
+	case *arrow.Uint32Type:
+		return "int unsigned"
+	case *arrow.Uint64Type:
+		return "bigint unsigned"
+	case *arrow.Float16Type, *arrow.Float32Type:
 		return "float"
-	case schema.TypeUUID:
+	case *arrow.Float64Type:
+		return "double"
+	case *arrow.Decimal128Type:
+		return fmt.Sprintf("decimal(%d,%d)", v.Precision, v.Scale)
+	case *arrow.Decimal256Type:
+		return fmt.Sprintf("decimal(%d,%d)", v.Precision, v.Scale)
+	case *types.UUIDType:
 		return "binary(16)"
-	case schema.TypeByteArray:
+	case *arrow.BinaryType, *arrow.LargeBinaryType, *arrow.FixedSizeBinaryType:
 		return "blob"
```

Comment:
Do we want to make this account for the fixes binary width?
```suggestion
	case *arrow.BinaryType, *arrow.LargeBinaryType:
		return "blob"
	case *arrow.FixedSizeBinaryType:
		return "blob(" + strconv.Itoa(v.ByteWidth) + ")"
```

Reply:
np, just wanted to check in with you on this feature.
Maybe `binary(n)` would work here better?